sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusSenderClientTest.java

```diff
@@ -0,0 +1,229 @@
+        MockitoAnnotations.initMocks(this);
+        when(asyncSender.getEntityPath()).thenReturn(ENTITY_NAME);
+        when(asyncSender.getFullyQualifiedNamespace()).thenReturn(NAMESPACE);
+
+        sender = new ServiceBusSenderClient(asyncSender, retryOptions.getTryTimeout());
+
+    }
+
+    @AfterEach
+    void teardown() {
+        sender.close();
+        singleMessageCaptor = null;
+        messageBatchCaptor = null;
+        Mockito.framework().clearInlineMocks();
+    }
+
+    @Test
+    void verifyProperties() {
+        Assertions.assertEquals(ENTITY_NAME, sender.getEntityPath());
+        Assertions.assertEquals(NAMESPACE, sender.getFullyQualifiedNamespace());
+    }
+
+    /**
+     * Verifies that an exception is thrown when we create a batch with null options.
+     */
+    @Test
+    void createBatchNull() {
+        Assertions.assertThrows(NullPointerException.class, () -> sender.createBatch(null));
+    }
+
+    /**
+     * Verifies that the default batch is the same size as the message link.
+     */
+    @Test
+    void createBatchDefault() {
+        // Arrange
+        ServiceBusMessageBatch batch =  new ServiceBusMessageBatch(MAX_MESSAGE_LENGTH_BYTES, null, null,
+            null);
+        when(asyncSender.createBatch()).thenReturn(Mono.just(batch));
+
+        //Act
+        ServiceBusMessageBatch batchMessage = sender.createBatch();
+
+        //Assert
+        Assertions.assertEquals(MAX_MESSAGE_LENGTH_BYTES, batchMessage.getMaxSizeInBytes());
+        Assertions.assertEquals(0, batchMessage.getCount());
+        verify(asyncSender, times(1)).createBatch();
+
+    }
+
+    /**
+     * Verifies we cannot create a batch if the options size is larger than the link.
+     */
+    @Test
+    void createBatchWhenSizeTooBigThanOnSendLink() {
+        // Arrange
+        int maxLinkSize = 1024;
+        int batchSize = maxLinkSize + 10;
+
+        // This event is 1024 bytes when serialized.
+        final CreateBatchOptions options = new CreateBatchOptions().setMaximumSizeInBytes(batchSize);
+        when(asyncSender.createBatch(options)).thenThrow(new IllegalArgumentException("too large size"));
+
+        // Act & Assert
+        try {
+            sender.createBatch(options);
+            Assertions.fail("Should not have created batch because batchSize is bigger than the size on SenderLink.");
+        } catch (Exception ex) {
+            Assertions.assertTrue(ex instanceof IllegalArgumentException);
+        }
+
+        verify(asyncSender, times(1)).createBatch(options);
+    }
+
+    /**
+     * Verifies that the producer can create a batch with a given {@link CreateBatchOptions#getMaximumSizeInBytes()}.
+     */
+    @Test
+    void createsMessageBatchWithSize() {
+        // Arrange
+        int maxLinkSize = 10000;
+        int batchSize = 1024;
+
+        // Overhead when serializing an event, to figure out what the maximum size we can use for an event payload.
+        int eventOverhead = 46;
```

Comment:
You don't need all this logic. You can assert that the correct properties for the batch were passed to asyncClient. We assume asyncClient does the right thing